pkg/store/bucket.go

```diff
@@ -996,44 +996,43 @@ func int64index(s []int64, x int64) int {
 }
 
 // getFor returns a time-ordered list of blocks that cover date between mint and maxt.
-// Blocks with the lowest resolution possible but not lower than the given resolution are returned.
-func (s *bucketBlockSet) getFor(mint, maxt, minResolution int64) (bs []*bucketBlock) {
-	if mint == maxt {
-		return nil
-	}
-
+// The covering is greedy, preferring blocks of the lowest allowed resolutions.
+func (s *bucketBlockSet) getFor(minTime, maxTime, minResolution, maxResolution int64) (bs []*bucketBlock) {
```

Comment:
We change a big for this part, so wonder how it will look like after rebase, but generally it looks like something we want :+1:

Reply:
👍  My changes are very small now actually.  Previously I had refactored this function to not be recursive, but I don't think that it helpful any more (the recursion does a nice job of filling in the gaps).